addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,252 @@
+import de.zazaz.iot.bosch.indego.IndegoController;
+import de.zazaz.iot.bosch.indego.IndegoException;
+
+/**
+ * The {@link BoschIndegoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jonas Fleck - Initial contribution
+ */
+public class BoschIndegoHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(BoschIndegoHandler.class);
+    private int commandToSend;
+
+    private Thread pollingThread;
+    private boolean running;
+
+    public BoschIndegoHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // Currently manual refrshing is not possible in the moment
+            return;
+        } else if (channelUID.getId().equals(STATE) && command instanceof DecimalType) {
+            if (command instanceof DecimalType) {
+                synchronized (this) {
+                    commandToSend = ((DecimalType) command).intValue();
+                    this.notifyAll();
+                }
+            } else {
+            }
+
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+        }
+    }
+
+    @Override
+    public void updateConfiguration(Configuration configuration) {
+        // TODO Auto-generated method stub
+        super.updateConfiguration(configuration);
+
+    }
+
+    private synchronized void poll() throws InterruptedException {
+        // Create controller instance
+        try {
+            IndegoController controller = new IndegoController(getConfig().get("username").toString(),
+                    getConfig().get("password").toString());
+            // Connect to server
+            controller.connect();
+            // Query the device state
+            DeviceStateInformation state = controller.getState();
+            DeviceStatus statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+            int eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+            int mowed = state.getMowed();
+            int error = state.getError();
+            int statecode = state.getState();
+            boolean ready = isReadyToMow(state.getState(), state.getError());
+
+            if (verifyCommand(commandToSend, eshStatus, state.getState(), error)) {
+                logger.debug("Sending command...");
+                updateState(TEXTUAL_STATE, new StringType("Refreshing..."));
+                controller.sendCommand(getCommandFromEshStatus(commandToSend));
+                commandToSend = 0;
+                try {
+                    for (int i = 0; i < 30 && !Thread.interrupted(); i++) {
```

Comment:
Maybe its possible to keep the connection alive but Bosch allows only one connection at the same time. Keeping the connection alive means that the user isn't able to use the regular Indego Smartphone App.  Also the mower only refreshes its state round about every 20 minutes. With the current code the user can choose the refresh interval. I use a refresh interval of 10 minutes which is absolutely fast enough. (Minimum is 1 minute. We don't want to annoy the Bosch engineers :-) ). 

The poll method handles commands too because the workflow is very similar. Before sending a command you need to get the current state (poll) of the mower to decide if the command makes sense. Then you can send the command and update the state. you can't know the state before sending the command. Maybe the mower decides to make a boarder cut or has an error and so on. So sending a command includes the same steps like a regular poll. 

I just wanted to make the code compact :)